notebooks/advanced_RAG.ipynb

```diff
@@ -0,0 +1,1101 @@
+{
```

Comment:
<p>It would be nice to say a few words about alternatives to FAISS, and why we choose it. </p><p>Maybe also a TLDR on cosine similarity an alternatives </p><p>I know this is covered in the link that you provide, but a couple of sentences here won't hurt. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/6/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Reply:
<p>On FAISS, I chose it because it works well. To my knowledge, the most important difference is speed. There is a benchmark <a href="https://ann-benchmarks.com/" target="_blank">here</a>, but not sure it is really informative given the plethora of choices and different evaluation settings.</p><p> </p><p>I added a few sentences about alternative distances.</p>